### PR TITLE
Add: hiddenActivityBanners

### PR DESCRIPTION
**Pull request by borislav-itskov**
Add: hiddenActivityBanners to activity.ts to allow for manually hiding the activity banner

This is the main change:
```
// do not show a banner for forcefully hidden banners
this.broadcastedButNotConfirmed
.filter(
  (op) =>
    !this.#hiddenActivityBanners.includes(
      `${op.accountAddr}-${op.networkId}-${op.timestamp}`
    )
)
...
```

### CODE REVIEW

**Review by Ivshti — APPROVED** [review]

File: src/interfaces/banner.ts [hunk 1]
```diff
@@ -79,6 +79,11 @@ export type Action =
       actionName: 'close-swap-and-bridge'
       meta: { activeRouteId: number }
     }
+  | {
+      label: 'Close'
+      actionName: 'hide-activity-banner'
+      meta: { timestamp: number; addr: string; network: string; isReject: boolean }
```
Comment (PetromirDev) [inline src/interfaces/banner.ts:85]: Three terms are used for the same action:
```
close- in the label ('Close')
reject- in meta (isReject: boolean)
hide- in the action (ACTIVITY_CONTROLLER_HIDE_BANNER)
```
please choose one term and stick to it for clarity or one term for internal use and one for the UI. For example `hide` in the code and `Close` in the UI

**Review by PetromirDev — CHANGES_REQUESTED** [review]

File: src/controllers/activity/activity.ts [hunk 2]
```diff
@@ -506,6 +510,15 @@ export class ActivityController extends EventEmitter {
     this.emitUpdate()
   }
 
+  hideBanner({ addr, network, timestamp }: { addr: string; network: string; timestamp: number }) {
+    this.#hiddenActivityBanners.push(`${addr}-${network}-${timestamp}`)
+
+    /* eslint-disable @typescript-eslint/no-floating-promises */
+    this.#storage.set('hiddenActivityBanners', this.#hiddenActivityBanners)
```
Comment (PetromirDev) [inline src/controllers/activity/activity.ts:517]: How long should these be stored for? Shouldn't there be a cleaning mechanism?

**Review by PetromirDev — CHANGES_REQUESTED** [review]

File: src/controllers/activity/activity.ts [hunk 3]
```diff
@@ -506,6 +510,15 @@ export class ActivityController extends EventEmitter {
...
+
+    this.emitUpdate()
```
[the remark this answers is not in the document]
Comment (borislav-itskov) [reply src/controllers/activity/activity.ts:519]: Okay, but I don't understand why should we do that? For example, in `removeAccountData`, we do the same as here: no awaits, first setting the storage and only after that updating

File: src/controllers/activity/activity.ts (hunk 2 above)
Comment (borislav-itskov) [reply src/controllers/activity/activity.ts:517]: Isn't it a bit overkill to implement a cleaning mechanism for this? It's not like you're going to rage hit "close" on every single banner, not to mention that most of the time you will not be able to see it them at all as the txn will finish before actually seeing the activity

Comment (borislav-itskov) [reply src/controllers/activity/activity.ts]: Decided offline: to change the logic completely

**Pull request merged**